test/demo/contextChannels.test.ts

```diff
@@ -0,0 +1,270 @@
+import 'jest';
+import {connect, Fin, Identity, Application} from 'hadouken-js-adapter';
+
+import * as fdc3Remote from './utils/fdc3RemoteExecution';
+
+const testManagerIdentity = {uuid: 'test-app', name: 'test-app'};
+
+const testContext = {type: 'test-context', name: 'contextName1', id: {name: 'contextID1'}};
+
+const startedApps:Application [] = [];
+
+let fin: Fin;
+
+beforeAll(async () => {
+    fin = await connect({address: `ws://localhost:${process.env.OF_PORT}`, uuid: 'TEST-contextChannels.ts'});
+    await expect(fin.Application.wrapSync({uuid: 'test-app', name: 'test-app'}).isRunning()).resolves.toBe(true);
+});
+
+afterEach(async () => {
+    jest.clearAllMocks();
+
+    for (const app of startedApps) {
+        await app.quit().catch(() => {});
+    }
+
+    startedApps.length = 0;
+});
+
+// Creates one window for each channel, and has that window join that channel if not undefined
+async function setupWindows(...channels: (string|undefined)[]): Promise<Identity[]> {
+    const app1 = {uuid: 'test-app-1', name: 'test-app-1'};
+    const app2 = {uuid: 'test-app-2', name: 'test-app-2'};
+    const app3 = {uuid: 'test-app-3', name: 'test-app-3'};
+    const app4 = {uuid: 'test-app-4', name: 'test-app-4'};
+
+    const appIdentities = [app1, app2, app3, app4];
+
+    // Creating apps takes time, so increase timeout
+    jest.setTimeout(channels.length * 5000);
+
+    const result: Identity[] = await Promise.all(channels.map(async (channel, index) => {
+        const identity = appIdentities[index];
```

Comment:
Is there always at-most 4 channels? Rather than pre-defining the list of identities, can this be: ?
```
const identity = {uuid: `test-app-${index}`, name: `test-app-${index}`};
```

Reply:
Maybe we'll have tests in here in future that use more than 4 windows, but we can set those test apps up when we need to (or we'll have some kind of dynamic directory/manifest thing by then).

I always feel iffy about treating IDs as anything other than totally opaque, and in practice the difference is going to be whether we get an error at `const identity = appIdentities[index]` or whether we get it at `await fdc3Remote.open(testManagerIdentity, identity.uuid)`, which is kinda neither here nor there.